tumblr/kanvas-ios
# Adds NSSecureCoding support for frame serialization

This PR adds:

* Conformances to `NSSecureCoding` protocols for movable views (drawing will come later).
* Saving these object archives to memory when switching between frames.
* Saving these objects to disk on export.

It does not handle GIFs and only saves content when `multipleExports` is used.

WordPress iOS PR https://github.com/wordpress-mobile/WordPress-iOS/pull/15902 contains another usage example and further illustrates how this functionality will be used.

## Testing

* Enable the `Camera Metal`, `Metal Filters`, `Scale Media to Fill` and `Multiple Exports` settings. Disable the `Camera OpenGL`, `Camera OpenGL Capture`, `New Camera Modes`, and `Export Stop Motion as Video` settings.
* Change the camera mode at the top to "Photo"
* Take photos / videos or select from your library
* Add **text** and **stickers** to photos or videos
* Ensure that switching between frames keeps the **text** and **stickers** contents.
* Export the photos and videos and ensure the added overlays appear in the exported media.

## Changes

### NSSecureCoding

The following files are new or existing ones which simply implement `NSSecureCoding`:
* Archive.swift
* StylableImageView.swift
* MovableView.swift
* MovableViewInnerElement.swift
* ViewTransformations.swift
* StylableTextView.swift

`MediaArchiver.swift` adds a new `archive` method to handle archiving exported media.
`CameraController.swift` adds several helper methods to retrieve types for archiving and a `show(media:)` method which is the entry point for showing media which has been saved to an archive.
`MultiEditorViewController` also receives several helper methods to handle archiving/unarchiving when switching between frames.
`CameraSegmentHandler` receives a few changes to help with the archival logic for image and video content.
